indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexPhaseRunner.java

```diff
@@ -472,4 +474,18 @@ int getAndIncrementNextSpecId()
   {
     return nextSpecId++;
   }
+
+  private long getSubtaskTimeoutMillisFromContext()
+  {
+    if (context.isEmpty()) {
+      return Tasks.DEFAULT_SUB_TASK_TIMEOUT_MILLIS;
+    }
+
+    Object raw = context.getOrDefault(
+            Tasks.SUB_TASK_TIMEOUT_KEY,
+            Tasks.DEFAULT_SUB_TASK_TIMEOUT_MILLIS
+    );
+
+    return (raw instanceof Number) ? ((Number) raw).longValue() : Tasks.DEFAULT_SUB_TASK_TIMEOUT_MILLIS;
```

Comment:
We should throw an exception if the given value is not a Long.
So, we can probably just simplify this:

```suggestion
    return (long) context.getOrDefault(
            Tasks.SUB_TASK_TIMEOUT_KEY,
            Tasks.DEFAULT_SUB_TASK_TIMEOUT_MILLIS
    );
```

Reply:
Sure, that would work too.